packages/blade/src/components/Tour/TourMask.tsx

```diff
@@ -0,0 +1,149 @@
+/* eslint-disable react/jsx-no-useless-fragment */
+import React from 'react';
+import type { FlattenSimpleInterpolation } from 'styled-components';
+import styled, { css, keyframes } from 'styled-components';
+import usePresence from 'use-presence';
+import type { TourMaskRect } from './types';
+import { useWindowSize } from '~utils/useWindowSize';
+import { makeSpace, useTheme } from '~utils';
+import { makeMotionTime } from '~utils/makeMotionTime';
+import { metaAttribute, MetaConstants } from '~utils/metaAttribute';
+
+const scaleIn = keyframes`
+  from {
+    fill-opacity: 0;
+  }
+  to {
+    fill-opacity: 1;
+  }
+`;
+const fadeOut = keyframes`
+  from {
+    fill-opacity: 1;
+  }
+  to {
+    fill-opacity: 0;
+  }
+`;
+
+const AnimatedFade = styled.rect<{ animationType: FlattenSimpleInterpolation | null }>(
+  ({ animationType }) =>
+    animationType === null
+      ? ''
+      : css`
+          ${animationType}
+        `,
+);
+
+type FadeRectProps = React.ComponentProps<'rect'> & {
+  show: boolean;
+};
+const FadeRect = ({ show, children, ...rest }: FadeRectProps): React.ReactElement => {
+  const { theme } = useTheme();
+
+  const duration = theme.motion.duration.gentle;
+  const enter = css`
+    animation: ${scaleIn} ${makeMotionTime(duration)}
+      ${theme.motion.easing.entrance.effective as string};
+  `;
+
+  const exit = css`
+    animation: ${fadeOut} ${makeMotionTime(duration)}
+      ${theme.motion.easing.exit.effective as string};
+  `;
+
+  const { isMounted, isVisible } = usePresence(Boolean(show), {
+    transitionDuration: duration,
+    initialEnter: false,
+  });
+
+  return (
+    <>
+      {isMounted && (
+        // @ts-expect-error
+        <AnimatedFade animationType={isVisible ? enter : exit} {...rest}>
+          {children}
+        </AnimatedFade>
+      )}
+    </>
+  );
+};
+
+type TourMaskProps = {
+  padding: number;
+  size: TourMaskRect;
+  isTransitioning: boolean;
+};
+
+const absoluteFill = {
+  position: 'fixed',
+  top: 0,
+  left: 0,
+  bottom: 0,
+  right: 0,
+  // TODO: tokenize z-index values
+  zIndex: 100,
```

Comment:
is this the same as popover?

Reply:
Yeah. Popover & the Mask will be on the same level. Although I will check once if all the zIndexes work correctly, especially what happens if a modal is already opened.

Reply:
Tested it out, need to use higher zIndex than modal. Since if modal is the highest 1000 zIndex the mask overlay should always be on top of it. 

Made the zIndex to
```
const tourMaskZIndex = modalHighestZIndex + 100;
```

<img width="691" alt="image" src="https://github.com/razorpay/blade/assets/35374649/294d9768-b232-4022-9a55-03cb050b713a">